pkg/service/schema/service.go

```diff
@@ -81,157 +78,56 @@ func NewSchemaService(config config.SchemaServiceConfig, s storage.ServiceStorag
 
 // CreateSchema houses the main service logic for schema creation. It validates the input, and
 // produces a schema value that conforms with the VC JSON SchemaID specification.
-// TODO(gabe) support data integrity proof generation on schemas, versioning, and more
+// TODO(gabe) support data integrity proofs for credential schemas
 func (s Service) CreateSchema(ctx context.Context, request CreateSchemaRequest) (*CreateSchemaResponse, error) {
 	logrus.Debugf("creating schema: %+v", request)
 
 	if !request.IsValid() {
 		return nil, sdkutil.LoggingNewErrorf("invalid create schema request: %+v", request)
 	}
 
-	schemaBytes, err := json.Marshal(request.Schema)
+	// validate the schema
+	jsonSchema := request.Schema
+	schemaBytes, err := json.Marshal(jsonSchema)
 	if err != nil {
 		return nil, sdkutil.LoggingErrorMsg(err, "could not marshal schema in request")
 	}
 	if err = schemalib.IsValidJSONSchema(string(schemaBytes)); err != nil {
 		return nil, sdkutil.LoggingErrorMsg(err, "provided value is not a valid JSON schema")
 	}
-
-	// create schema
-	schemaID := uuid.NewString()
-	schemaValue := schema.VCJSONSchema{
-		Type:     schema.VCJSONSchemaType,
-		Version:  Version1,
-		ID:       schemaID,
-		Name:     request.Name,
-		Author:   request.Author,
-		Authored: time.Now().Format(time.RFC3339),
-		Schema:   prepareJSONSchema(schemaID, request.Name, request.Schema),
+	if !schema.IsSupportedJSONSchemaVersion(jsonSchema.Schema()) {
+		return nil, sdkutil.LoggingNewErrorf("unsupported schema version: %s", jsonSchema.Schema())
+	}
+	if jsonSchema.ID() != "" {
+		logrus.Infof("schema has id: %s, which is being overwritten", jsonSchema.ID())
 	}
 
-	storedSchema := StoredSchema{ID: schemaID, Schema: schemaValue}
-
-	// sign the schema
-	if request.Sign {
-		signedSchema, err := s.signSchemaJWT(ctx, request.AuthorKID, schemaValue)
-		if err != nil {
-			return nil, sdkutil.LoggingError(err)
+	// set id, name, and description on the schema
+	schemaID := uuid.NewString()
+	jsonSchema[schema.JSONSchemaIDProperty] = strings.Join([]string{s.Config().ServiceEndpoint, schemaID}, "/")
+	if jsonSchema[schema.JSONSchemaNameProperty] != "" && request.Name != "" {
+		logrus.Infof("schema has name: %s, which is being overwritten", jsonSchema[schema.JSONSchemaNameProperty])
+	}
+	jsonSchema[schema.JSONSchemaNameProperty] = request.Name
+	if request.Description != "" {
+		if jsonSchema[schema.JSONSchemaDescriptionProperty] != "" {
+			logrus.Infof("schema has description: %s, which is being overwritten", jsonSchema[schema.JSONSchemaDescriptionProperty])
 		}
-		storedSchema.SchemaJWT = signedSchema
+		jsonSchema[schema.JSONSchemaDescriptionProperty] = request.Description
 	}
 
+	// TODO(gabe) support signing credential schemas
+	// create schema
+	storedSchema := StoredSchema{ID: schemaID, Schema: jsonSchema}
```

Comment:
There are two IDs within a StoredSchema: `id` and `schema.$id`. I think it's easily confused. Would it make sense to only have one?

Reply:
the reason for two: when you view the schema on it's own you need a fully qualified URI to resolve it (e.g. `tbd.website/schemas/01841408-1408140148`

when you are talking to the service, it is awkward to request anything but `01841408-1408140148`

so `01841408-1408140148` is the service identifier, whereas the `$id` is the fully qualified identifier